src/mod/asr_tts/mod_unimrcp/mod_unimrcp.c

```diff
@@ -1913,6 +1918,27 @@ static apt_bool_t speech_on_channel_add(mrcp_application_t *application, mrcp_se
 		} else {
 			switch_event_add_header_string(event, SWITCH_STACK_BOTTOM, "MRCP-Resource-Type", "ASR");
 		}
+		switch_event_add_header_string(event, SWITCH_STACK_BOTTOM, "MRCP-Session-Identifier", schannel->unimrcp_session->id.buf);
```

Comment:
Validate if id is empty unless you are sure that at this point id.buf always contain a valid pointer